### PR TITLE
Review protocol 2.1

### PR DESCRIPTION
**Pull request by nmatthews-witekio**
DON'T MERGE

Review comments on "protocol-2.1".

UNTESTED / UNREVIEWED.

### CODE REVIEW

File: tests/discovery.c [hunk 1]
```diff
@@ -18,7 +18,8 @@ static void report_sync_error(IotclSyncResponse *response, const char* sync_resp
         printf("IOTC_SyncResponse is NULL. Out of memory?\n");
         return;
     }
-    switch(response->ds) {
+    // FIXME TODO CHECK -- field is renamed?
```
Comment (nik-markovic) [inline tests/discovery.c:21]: I don't think that Neerav has updated tests at all, and he should have. You probably don't want to be writing or reviewing this this test code until it is ready. Neerav should know how to properly unit-test the code. There should be more to these tests than just fixing compile errors and matching some variable name changes. There's the hello message etc.

File: src/iotconnect_discovery.c [hunk 2]
```diff
@@ -76,6 +76,7 @@ IotclDiscoveryResponse *iotcl_discovery_parse_discovery_response(const char *res
         goto cleanup;
     }
 
+    if(base_url_cjson != NULL)
```
Comment (nik-markovic) [inline src/iotconnect_discovery.c:79]: Minor: The if opening curly brace should be in the same line as if. Block comment should be removed.

**Review by nik-markovic — APPROVED** [review]

**Comment by nmatthews-witekio** [comment]
Out of date... need to rerun cppcheck on the new code.

cppcheck --enable=all --force --inconclusive . 

and grep for "]".

**Comment by nmatthews-witekio** [comment]
Transcribe comments into PR#15 and close

**Comment by nmatthews-witekio** [comment]
Closing -- as have transcribed comments into a direct review of https://github.com/avnet-iotconnect/iotc-c-lib/pull/15

**Pull request closed without merging**